org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/catalog/XMLCatalogDiagnosticsParticipant.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lemminx.extensions.catalog;
+
+import java.text.MessageFormat;
+import java.util.List;
+
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.extensions.contentmodel.settings.XMLValidationSettings;
+import org.eclipse.lemminx.services.extensions.diagnostics.IDiagnosticsParticipant;
+import org.eclipse.lemminx.utils.DOMUtils;
+import org.eclipse.lemminx.utils.FilesUtils;
+import org.eclipse.lemminx.utils.URIUtils;
+import org.eclipse.lemminx.utils.XMLPositionUtility;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+
+/**
+ * Validate XML catalog.
+ *
+ */
+public class XMLCatalogDiagnosticsParticipant implements IDiagnosticsParticipant {
+
+	private static final String ERROR_STRING = "The file ''{0}'' cannot be found.";
+
+	@Override
+	public void doDiagnostics(DOMDocument xmlDocument, List<Diagnostic> diagnostics,
+			XMLValidationSettings validationSettings, CancelChecker monitor) {
+		if (!DOMUtils.isCatalog(xmlDocument)) {
+			return;
+		}
+
+		for (CatalogEntry catalogEntry : CatalogUtils.getCatalogEntries(xmlDocument)) {
+			String path = CatalogUtils.getResolvedLocation(xmlDocument, catalogEntry);
+			if (!FilesUtils.isValidPath(FilesUtils.getPath(path)) && URIUtils.isFileResource(path)) {
```

Comment:
If I understand correctly, it means that CatalogUtils.getResolvedLocation returns a path with `file://` scheme even if the  original uri doesn't starts with file://

If it that could you add a comment here with this information.

Reply:
Yes that's correct

Reply:
Ok thanks,could you add a comment here with a sample too.